rmgpy/data/solvation.py

```diff
@@ -1844,25 +1919,83 @@ def get_Kfactor(self, solute_data, solvent_data, T):
-        """
-        Given solute_data, solvent_data, and temperature, calculates the Gibbs free energy of
-        solvation at T if the solvent's name_in_coolprop is not None.
+    def get_T_dep_solvation_energy_from_LSER_298(self, solute_data, solvent_data, T):
+        """Returns solvation free energy and K-factor for the input temperature based on the 298 K solvation properties
+          calculated from the LSER method.
+
+        Args:
+            solute_data: :class:`SoluteData` object `solute_data`.
+            solvent_data: :class:`SolventData` object `solvent_data`.
+            T (float): input temperature in K.
+
+        Returns:
+            delG (float): solvation free energy at the input temperature in J/mol.
+            Kfactor (float): K-factor at the input temperature. K-factor is defined as a ratio of the mole fraction
+            of a solute in a gas-phase to the mole fraction of a solute in a liquid-phase at equilibrium.
+
+        Raises:
+            DatabaseError: if `solute_data.name_in_coolprop` is None or `solute_data` has any missing Abarham or
+            Mintz solvent parameters.
+
         """
-        Kfactor = self.get_Kfactor(solute_data, solvent_data, T)
-        rho_g = PropsSI('Dmolar', 'T', T, 'Q', 1, solvent_data.name_in_coolprop) # saturated gas phase density of the solvent, in mol/m^3
-        rho_l = PropsSI('Dmolar', 'T', T, 'Q', 0, solvent_data.name_in_coolprop) # saturated liquid phase density of the solvent, in mol/m^3
+
+        solvent_name = solvent_data.name_in_coolprop
+        # check whether all solvent parameters exist
+        solvent_param_list = [solvent_data.s_g, solvent_data.b_g, solvent_data.e_g,
+                              solvent_data.l_g, solvent_data.a_g, solvent_data.c_g,
+                              solvent_data.s_h, solvent_data.b_h, solvent_data.e_h,
+                              solvent_data.l_h, solvent_data.a_h, solvent_data.c_h,
+                              ]
+        param_found_list = [param is not None for param in solvent_param_list]
+
+        if solvent_name is not None and all(param_found_list):
+            correction = self.get_solvation_correction(solute_data, solvent_data)
+            delG298 = correction.gibbs  # in J/mol
+            delH298 = correction.enthalpy  # in J/mol
+            delS298 = correction.entropy  # in J/mol/K
+        else:
+            raise DatabaseError("K-factor parameter calculation is not available for the solvent "
+                                "whose `name_in_coolprop` is None or that is missing the Abraham and Mintz "
+                                "solvent parameters.")
+
+        Kfactor = self.get_Kfactor(delG298, delH298, delS298, solvent_name, T)
+        rho_g = get_gas_saturation_density(solvent_name, T)
+        rho_l = get_liquid_saturation_density(solvent_name, T)
         delG = constants.R * T * math.log(Kfactor * rho_g / (rho_l))  # in J/mol
-        return delG
+        return delG, Kfactor
```

Comment:
I think you can replace all of this with `return get_T_dep_solvation_energy_from_input_298(delG298, delH298, delS298, solvent_name, T)`, but correct me if I am wrong

Reply:
You are correct! I replaced it all with that line in this fixup commit f522be9.